app/oembed/server/server.js

```diff
@@ -108,17 +108,26 @@ const getUrlContent = async function (urlObj, redirectCount = 5) {
 			'User-Agent': `${settings.get('API_Embed_UserAgent')} Rocket.Chat/${Info.version}`,
 			'Accept-Language': settings.get('Language') || 'en',
 		},
-		size: 250000, // max size of the response body
+		size: sizeLimit, // max size of the response body, this was not working as expected so I'm also manually verifying that on the iterator
 		...(settings.get('Allow_Invalid_SelfSigned_Certs') && {
 			agent: getUnsafeAgent(parsedUrl.protocol),
 		}),
 	});
 
+	let totalSize = 0;
+	const chunks = [];
 	for await (const chunk of response.body) {
+		totalSize += chunk.length;
 		chunks.push(chunk);
+
+		if (totalSize > sizeLimit) {
+			SystemLogger.info({ msg: 'OEmbed request size exceeded', url });
+			break;
```

Comment:
I didn't found any issue on breaking the async iterator loop this way.
not sure if this could produce a memory though